gms/gossiper.cc

```diff
@@ -317,7 +317,7 @@ future<> gossiper::handle_ack_msg(msg_addr id, gossip_digest_ack ack_msg) {
     assert(_msg_processing > 0);
 
     return f.then([this, from = id, ack_msg_digest = std::move(g_digest_list), mp = std::move(mp), g = this->shared_from_this()] () mutable {
-        if (this->is_in_shadow_round()) {
+        if (this->is_in_shadow_round() || _in_long_shadow_round) {
```

Comment:
This is confusing since if the gossiper is not `_in_shadow_round` then there's no need to call `finish_shadow_round`.
It'd be better to leave this unchanged and add:
```
if (_in_long_shadow_round) {
    // explain why we return early
    return make_ready_future<>();
}
```

Reply:
Ok